### ISSUE TITLE
Directly implementing gears on scRNA-seq data

### ISSUE THREAD

**Issue by XiaoMi93** [snap-stanford/GEARS#53]
Hi,
Thank you for publishing such an excellent paper! I'm new to perturb-seq and only have single cell transcriptome data. I wonder whether I can use gears directly on the scRNA-seq data to infer the perturbation. Or I must provide my own perturb-seq data.
Thanks!

**Comment by XiaoMi93** [comment]
As I only have scRNA-seq data, so I set the condition to ctrl for all cells during step 2 in the data_tutorial.ipynb file ((2) Create your own Perturb-Seq data). I got this error:
AttributeError                            Traceback (most recent call last)
Cell In[50], [line 7](vscode-notebook-cell:?execution_count=50&line=7)
----> [7](vscode-notebook-cell:?execution_count=50&line=7) pert_data.new_data_process(dataset_name = scRNA', adata = adata) # specific dataset name and adata object

File [~/anaconda3/lib/python3.8/site-packages/gears/pertdata.py:250), in PertData.new_data_process(self, dataset_name, adata, skip_calc_de)
    [248](https://file+.vscode-resource.vscode-cdn.net/python3.8/site-packages/gears/pertdata.py:248)     os.mkdir(save_data_folder)
    [249](https://file+.vscode-resource.vscode-cdn.net/python3.8/site-packages/gears/pertdata.py:249) self.dataset_path = save_data_folder
--> [250](https://file+.vscode-resource.vscode-cdn.net/site-packages/gears/pertdata.py:250) self.adata = get_DE_genes(adata, skip_calc_de)
    [251](https://file+.vscode-resource.vscode-cdn.net/site-packages/gears/pertdata.py:251) if not skip_calc_de:
    [252](https://file+.vscode-resource.vscode-cdn.net/python3.8/site-packages/gears/pertdata.py:252)     self.adata = get_dropout_non_zero_genes(self.adata)

File [~/anaconda3/envs/Tres/lib/python3.8/site-packages/gears/data_utils.py:64](https://file+.vscode-resource.vscode-cdn.net/python3.8/site-packages/gears/data_utils.py:64), in get_DE_genes(adata, skip_calc_de)
     [62](https://file+.vscode-resource.vscode-cdn.net/gears/data_utils.py:62) adata.obs = adata.obs.astype('category')
     [63](https://file+.vscode-resource.vscode-cdn.net/gears/data_utils.py:63) if not skip_calc_de:
---> [64](https://file+.vscode-resource.vscode-cdn.net/gears/data_utils.py:64)     rank_genes_groups_by_cov(adata, 
     [65](https://file+.vscode-resource.vscode-cdn.net/gears/data_utils.py:65)                      groupby='condition_name', 
     [66](https://file+.vscode-resource.vscode-cdn.net/gears/~/anaconda3/envs/Tres/lib/python3.8/site-packages/gears/data_utils.py:66)                      covariate='cell_type', 
     [67](https://file+.vscode-resource.vscode-cdn.net/gears/~/anaconda3/envs/Tres/lib/python3.8/site-packages/gears/data_utils.py:67)                      control_group='ctrl_1', 
     [68](https://file+.vscode-resource.vscode-cdn.net/gears/~/anaconda3/envs/Tres/lib/python3.8/site-packages/gears/data_utils.py:68)                      n_genes=len(adata.var),
     [69](https://file+.vscode-resource.vscode-cdn.net/gears/~/anaconda3/envs/Tres/lib/python3.8/site-packages/gears/data_utils.py:69)                      key_added = 'rank_genes_groups_cov_all')
...
    [631](https://file+.vscode-resource.vscode-cdn.net/gears/~/anaconda3/envs/Tres/lib/python3.8/site-packages/scanpy/tools/_rank_genes_groups.py:631)     'pvals_adj': 'float64',
    [632](https://file+.vscode-resource.vscode-cdn.net/gears/~/anaconda3/envs/Tres/lib/python3.8/site-packages/scanpy/tools/_rank_genes_groups.py:632) }
    [634](https://file+.vscode-resource.vscode-cdn.net/gears/~/anaconda3/envs/Tres/lib/python3.8/site-packages/scanpy/tools/_rank_genes_groups.py:634) for col in test_obj.stats.columns.levels[0]:

AttributeError: 'NoneType' object has no attribute 'columns'

I think this is because I only provide one condition. Does this mean that I can not directly implement Gears on non-perturb-seq data?

**Comment by yhr91** [comment]
Sorry for the delayed response!

It's hard to follow exactly which dataset you used for training but if I understand correctly you tried using non-perturb seq data to train GEARS. This will not work as GEARS needs perturbational data, ideally from a few different genetic perturbations. 

If I misunderstood your question, please feel free to re-open this issue.

**Comment by ManuelMoradiellos** [comment]
I too have the same question as @XiaoMi93!

I was wondering if one can use one of the pre-trained models (_on Perturb-seq data_) and apply them to scRNA-seq data to infer possible transcriptional response to a list of known/given perturbations; not to train the model per se but to apply the ones already trained on a different data source, despite them using different cell types and tissue just to inquire about possible similarities.

Thanks in advance!
